lib/protobuf/dsl.ex

```diff
@@ -226,19 +242,25 @@ defmodule Protobuf.DSL do
     end
   end
 
-  defp def_t_typespec(%MessageProps{enum?: true} = props, _extension_props) do
+  defp def_t_typespec(_props, _extension_props, transform_module_ast) when not is_nil(transform_module_ast) do
+    quote do
+      @type t() :: unquote(transform_module_ast).t(__MODULE__)
```

Comment:
The biggest potential issue is that we are assuming that the function ast will always be a module. But I can't see it being something else if you are using the library properly :)